### PR TITLE
WIP: feat: adding basic app structure with TS

### PR DESCRIPTION
**Pull request by MateusAndrade**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Lint has passed locally and any fixes were made for failures
- [ ] Code Coverage stayed the same or increased

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Tests related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/app/.watchmanconfig [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (viniarruda) [inline packages/app/.watchmanconfig]: remove or add conf

File: packages/app/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+ 
+{
+  "compilerOptions": {
```
Comment (viniarruda) [inline packages/app/tsconfig.json]: "include": ["src"],
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "target": "es6",
    "allowJs": true,
  }

add this

File: packages/app/.watchmanconfig (hunk 1 above)
Comment (MateusAndrade) [reply packages/app/.watchmanconfig]: Having this file is the default setup at rn

File: packages/app/tsconfig.json (hunk 2 above)
Comment (MateusAndrade) [reply packages/app/tsconfig.json]: We need to discuss how we are going to handle the versions of ts, native, etc. Actually the project are looking for reference at their root folder, but using a monorepo architecture will set-up all dependencies at the root folder.

**Pull request closed without merging**